tests/kvm.rs

```diff
@@ -0,0 +1,201 @@
+mod config;
+
+use env_logger;
+use log::debug;
+use std::mem;
+use std::panic;
+use std::process::{Command, Stdio};
+use std::sync::{mpsc, Once};
+use std::thread;
+use std::time::Duration;
+
+use config::{KVMI_SOCKET, TIMEOUT, VIRSH_URI, VM_NAME};
+
+// to init env logger
+static INIT: Once = Once::new();
+
+fn run_test<T>(test: T) -> ()
```

Comment:
What about: https://docs.rs/test-context/0.1.2/test_context/

Reply:
Interesting crate, but it doesn't seem well known in the community ?
the repo has 4 :star: and the last commit was 4 months ago, so I'm not convinced yet  :)

Reply:
Hm, still better than implementing everything yourself, I think. ;) You are right, it's not well known, but it looks decent to me at first glance (although the macros were a hard pass for me^^). It also has very little functionality which might explain why it is not actively developed. Moreover, it's not like we would be using it in production code.
I personally like how this crate let's you structure your test harness because it's more in line with what I am used to from well established test frameworks in other languages than the custom harness. Plus, it would be neat if you could use this together with ntest so you don't have to implement context cancellation yourself.

Reply:
To be clear, I totally understand if that's not convincing enough for you ;)